packages/toolboxes/src/tron/toolbox.ts

```diff
@@ -441,22 +424,26 @@ export const createTronToolbox = async (
     const { recipient, assetValue, memo, sender, expiration } = params;
     const isNative = assetValue.isGasAsset;
 
+    const addTxData = async (transaction: TronTransaction, memo?: string, expiration?: number) => {
+      const transactionWithMemo = memo
+        ? await tronWeb.transactionBuilder.addUpdateData(transaction, memo, "utf8")
+        : transaction;
+
+      const transactionFinal = expiration
+        ? await tronWeb.transactionBuilder.extendExpiration(transactionWithMemo, expiration)
+        : transactionWithMemo;
+
+      return transactionFinal;
+    };
+
     if (isNative) {
       const transaction = await tronWeb.transactionBuilder.sendTrx(
         recipient,
         assetValue.getBaseValue("number"),
         sender,
       );
 
-      if (memo) {
-        return tronWeb.transactionBuilder.addUpdateData(transaction, memo, "utf8");
-      }
-
-      if (expiration) {
-        tronWeb.transactionBuilder.extendExpiration(transaction, expiration);
-      }
-
-      return transaction;
+      return addTxData(transaction, memo, expiration);
```

Comment:
That part seems to be repetitive (called on every block). Consider handling that part as last like.

Additionally. We often name variables and return them for easier readability and debugging. So preferred way would be
```suggestion
      const txWithData = addTxData({ transaction, memo, expiration });
      return txWithData
```